examples/submit_samples.py

```diff
@@ -0,0 +1,304 @@
+    return file_paths
+
+
+def main():
+    """Submit all samples or hashes by downloading from VT first."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        dest="config_file",
+        default="./data/tau_clients.ini",
+        help="read config from here",
+    )
+    parser.add_argument(
+        "input_bits",
+        metavar="input_bits",
+        nargs="+",
+        help="input files, hashes, or list of hashes",
+    )
+    parser.add_argument(
+        "-b",
+        "--bypass-cache",
+        dest="bypass_cache",
+        action="store_true",
+        default=False,
+        help="whether to bypass the cache",
+    )
+    parser.add_argument(
+        "-t",
+        "--input-type",
+        choices=[
+            INPUT_TYPE_FILE_HASH,
+            INPUT_TYPE_FILE_SAMPLE,
+        ],
+        dest="input_type",
+        default=None,
+        help="input type",
+    )
+    args = parser.parse_args()
+    if not os.path.isfile(args.config_file):
+        raise ValueError(f"No configuration file at file path '{args.config_file}'")
+    conf = configparser.ConfigParser()
+    conf.read(args.config_file)
+
+    try:
+        vt_client = vt.Client(apikey=conf.get("vt", "apikey"))
+    except configparser.Error:
+        print("VT credentials not found. Hash submissions are disabled")
+        vt_client = None
+
+    # Parse and decode the input
+    file_paths = []
+    file_hashes = []
+    for input_bit in args.input_bits:
+        input_type = args.input_type
+        if not input_type:
+            input_type = detect_input_type(input_bit)
+            print(f"Input '{input_bit}' is treated as {input_type}. Force using '-t' option")
+        if input_type == INPUT_TYPE_FILE_HASH:
+            if not vt_client:
+                print(f"Hash submission '{input_bit}' requires VT credentials. Skipping.")
+                continue
+            if os.path.isfile(input_bit):
+                file_hashes.extend(get_file_hashes(input_bit))
+            elif is_hash(input_bit):
+                file_hashes.append(input_bit)
+        elif input_type == INPUT_TYPE_FILE_SAMPLE:
+            file_paths.extend(get_file_paths(input_bit))
+        else:
+            raise ValueError("Unknown input type")
+    print(f"Decoded input into {len(file_hashes)} file hashes and {len(file_paths)} samples")
+
+    # Submit
+    analysis_client = nsx_defender.AnalysisClient.from_conf(conf, "analysis")
+    submission_start_ts = analysis_client.get_api_utc_timestamp()
+    submissions = []
+    task_to_source = {}
+    for file_path in file_paths:
+        with open(file_path, "rb") as f:
+            ret = analysis_client.submit_file(f.read(), bypass_cache=args.bypass_cache)
+            submissions.append(ret)
+            task_to_source[ret["task_uuid"]] = file_path
+    for file_hash in file_hashes:
+        file_data = download_from_vt(vt_client, file_hash)
```

Comment:
Here `vt_client` might be None

Reply:
Not really, because if before was None file_hashes is empty here